Makefile

```diff
@@ -82,8 +82,10 @@ LDFLAGS=
 
 # some gcc versions treat var in `if (type var = val)` as unused
 # some gcc versions treat variables used in constexpr if blocks as unused
+# some gcc versions warn maybe uninitalized on optional<std::string> members on structs
 ifeq ($(findstring g++,$(shell $(CXX) --version)),g++)
 	CXXFLAGS+=-Wno-unused
+	CXXFLAGS+=-Wno-maybe-uninitialized
```

Comment:
in CI, we use the makefile, and on gcc we should ignore the maybe-uninitialized warning because it gives us a lot of false positives.